src/tools/ceph_objectstore_tool.cc

```diff
@@ -3601,7 +3601,16 @@ int main(int argc, char **argv)
     }
     return 1;
   }
-
+  auto umount_fs = make_scope_guard([&] {
```

Comment:
I'm afraid this commit isn't just a refactoring. In the very end we have an extra logic converting the return value:

```cpp
  if (ret < 0)
    ret = 1;
  return ret;
}
```

Would it take effect after the change?

Reply:
@rzarzynski  right. that part is difficult. i reverted this change. could you take another look?